src/webgl/p5.Camera.js

```diff
@@ -1638,6 +1638,324 @@ p5.Camera = class Camera {
+    // some axis. That's the angle. Letting this be theta, trace becomes 1+2cos(theta).
+    let cosTheta = 0.5 * (m[0] + m[4] + m[8] - 1);
+    // If the angle is close to 0, the two matrices are very close,
+    // so in that case we execute linearly interpolate.
+    if (1 - cosTheta < 0.0000001) {
+      // Obtain the front vector and up vector by linear interpolation
+      // and normalize them.
+      const lerpedFront = new p5.Vector(
+        (1 - amt) * m0[2] + amt * m1[2],
+        (1 - amt) * m0[5] + amt * m1[5],
+        (1 - amt) * m0[8] + amt * m1[8]
+      ).normalize();
+      const lerpedUp = new p5.Vector(
+        (1 - amt) * m0[1] + amt * m1[1],
+        (1 - amt) * m0[4] + amt * m1[4],
+        (1 - amt) * m0[7] + amt * m1[7]
+      ).normalize();
+
+      // set the camera
+      // The eye position and center position are calculated based on the front vector.
+      this.camera(
+        lerpedMediumX + ratio * lerpedDist * lerpedFront.x,
+        lerpedMediumY + ratio * lerpedDist * lerpedFront.y,
+        lerpedMediumZ + ratio * lerpedDist * lerpedFront.z,
+        lerpedMediumX + (ratio-1) * lerpedDist * lerpedFront.x,
+        lerpedMediumY + (ratio-1) * lerpedDist * lerpedFront.y,
+        lerpedMediumZ + (ratio-1) * lerpedDist * lerpedFront.z,
+        lerpedUp.x, lerpedUp.y, lerpedUp.z
+      );
+      return;
+    }
+
+    // Calculates the axis vector and the angle of the difference orthogonal matrix.
+    // The axis vector is what I explained earlier in the comments.
+    // similar calculation is here:
+    // https://github.com/mrdoob/three.js/blob/dev/src/math/Quaternion.js#L294
+    let a, b, c, sinTheta;
+    let invOneMinusCosTheta = 1 / (1 - cosTheta);
+    if (m[0] > m[4] && m[0] > m[8]) {
+      a = Math.sqrt((m[0] - cosTheta) * invOneMinusCosTheta); // not zero.
+      invOneMinusCosTheta /= a;
+      b = 0.5 * (m[1] + m[3]) * invOneMinusCosTheta;
+      c = 0.5 * (m[2] + m[6]) * invOneMinusCosTheta;
+      sinTheta = 0.5 * (m[7] - m[5]) / a;
+    } else if (m[4] > m[8]) {
+      b = Math.sqrt((m[4] - cosTheta) * invOneMinusCosTheta); // not zero.
+      invOneMinusCosTheta /= b;
+      c = 0.5 * (m[5] + m[7]) * invOneMinusCosTheta;
+      a = 0.5 * (m[1] + m[3]) * invOneMinusCosTheta;
+      sinTheta = 0.5 * (m[2] - m[6]) / b;
+    } else {
+      c = Math.sqrt((m[8] - cosTheta) * invOneMinusCosTheta); // not zero.
+      invOneMinusCosTheta /= c;
+      a = 0.5 * (m[2] + m[6]) * invOneMinusCosTheta;
+      b = 0.5 * (m[5] + m[7]) * invOneMinusCosTheta;
+      sinTheta = 0.5 * (m[3] - m[1]) / c;
+    }
+
+    // Constructs a new matrix after interpolating the angles.
+    // Multiplying mat0 by the first matrix yields mat1, but by creating a state
+    // in the middle of that matrix, you can obtain a matrix that is
+    // an intermediate state between mat0 and mat1.
+    const angle = amt * Math.atan2(sinTheta, cosTheta);
```

Comment:
ah I see what you mean about quaternions maybe making the implementation more difficult, since you mostly need the angle in order to then pick a different angle based on `amt`. Just for my understanding, I suppose you'd want to convert the quaternion to axis/angle, update the angle, then make a new quaternion from that, then apply that to m0, and this current implementation  avoids unnecessary work in making the intermediate states?

Reply:
I'm not familiar with quaternions in the first place, so I'm not sure. Orthogonal matrices are easier to work with, so I've thought about implementing them in that direction. It's very easy because we can get the difference just by taking the transpose and multiplying it. I thought it would be easier to implement this way because it avoids the extra effort of converting to quaternions.

Reply:
The reason I used the word quaternion was because I thought that it was doing something like that in the atmosphere while researching various things. I'm not sure how it would actually be implemented. However, I got the impression that it would take a lot of effort to convert between them.